README.md

```diff
@@ -105,7 +104,7 @@ The context object also varies depending on the value being decorated. Breaking
 - `isStatic`: Whether or not the value is a `static` class element. Only applies to class elements.
 - `isPrivate`: Whether or not the value is a private class element. Only applies to class elements.
 - `addInitializer`: Allows the user to add additional initialization logic. This is available for all decorators which operate per-class, as opposed to per-instance (in other words, decorators which do not have kind `"field"` - discussed in more detail below).
-- `setMetadata`: Allows the user to define some metadata to be associated with this property. This metadata can then be accessed on the class via `Symbol.metadata`. See the section on Metadata below for more details.
+- `addMetadata`: Allows the user to define some metadata to be associated with this property. This metadata can then be accessed on the class via `Symbol.metadata`. See the section on Metadata below for more details.
```

Comment:
they're not really associated with the property tho unless they're stored under the property name somehow - if it's just a bare array, then it's only associated with the class.

Reply:
Good point, updated

Reply:
There was also an implicit question here :-) why not have the metadata value be an object, whose keys are property names, and whose values are arrays? To anticipate the “what about the class, or collisions between static and instance” part, what about an object like this:
```js
{
  prototype: {
    foo: [],
  },
  static: {
    bar: [],
  },
  instance: {
    baz: [],
  },
  constructor: [],
}
```

Reply:
This wouldn't work for private metadata, which was the whole reason we had the divide between `public` and `private`. This gets into the territory of "it's too complex" which seems to be a sticking point here, so this proposal is to be as simple as possible. Users can of course `addMetadata` with an object which has the name of the property and then sort out the properties themselves.

Reply:
ah, true enough. sounds like that’s boilerplate every metadata producer would likely need.